google-tag-manager.html

```diff
@@ -22,41 +22,109 @@
-
-      _computeSrc: function(cid) {
-        return '//www.googletagmanager.com/ns.html?id=' + cid;
-      }
-    });
-  </script>
+    <script>
+        Polymer({
+            is: 'google-tag-manager',
+            timeout: false,
+            isAnalyticsLoaded: false,
+            offlineEvents: [],
+            properties: {
+                /**
+                 * `cid` is the container ID
+                 */
+                cid: {
+                    type: String,
+                }
+            },
+
+            // Element Lifecycle
+
+            attached: function() {
+                (function(w, d, s, l, i) {
+                    w[l] = w[l] || [];
+                    w[l].push({
+                        'gtm.start': new Date().getTime(),
+                        event: 'gtm.js'
+                    });
+                    var f = d.getElementsByTagName(s)[0],
+                        j = d.createElement(s),
+                        dl = l != 'dataLayer' ? '&l=' + l : '';
+                    j.async = true;
+                    j.src =
+                        '//www.googletagmanager.com/gtm.js?id=' + i + dl;
+                    f.parentNode.insertBefore(j, f);
+                })(window, document, 'script', 'dataLayer', this.cid);
+            },
+            _checkGA: function() {
+                if (typeof ga !== "function") {
+                    return;
+                }
+                clearInterval(this.interval);
+                this.isAnalyticsLoaded = true;
+                this._processOfflineEvents.bind(this)();
+            },
+
+            _processOfflineEvents: function() {
+                // is ga ok?
+                this.offlineEvents.map(function sendEvent(ev) {
+                    ga('send', 'event', ev.category, ev.event, ev.label);
+                });
+                //empty the array
+                this.offlineEvents.length = 0;
+            },
+
+            createLabel: function(category, event, sessionId, customerId, attributes) {
+
+                function guid() {
+                    function s4() {
+                        return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
+                    }
+                    return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+                        s4() + '-' + s4() + s4() + s4();
+                }
+
+                var id = guid();
+                var timestamp = new Date().getTime();
+                var label = id + '|' + timestamp + '|' + sessionId + '|' + customerId;
+                for (i = 0; i < attributes.length; i++) {
+                    label = label + '|' + attributes[i];
+                }
+                return label;
+            },
+
+            ready: function() {
+                window.track = function(category, eventName, sessionId, customerId, attributes) {
+
+                    var label = this.createLabel(category, eventName, sessionId, customerId, attributes);
```

Comment:
The `this` here feels a little overhead, maybe I'll include the fns in the same context instead :P